presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/credential/InlineCredentialProvider.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc.credential;
+
+import javax.inject.Inject;
+
+import static java.util.Objects.requireNonNull;
+
+public class InlineCredentialProvider
+        implements CredentialProvider
+{
+    private final CredentialConfig credentialsConfig;
```

Comment:
Typically we use configs during the initialization (not on runtime). So I would inline user and password as separate fields here, instead of using config directly.